polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -757,15 +757,22 @@ ValueCategory MLIRScanner::VisitUnaryOperator(clang::UnaryOperator *U) {
     assert(Sub.val);
     Value Val = Sub.getValue(Builder);
 
-    if (!isa<IntegerType>(Val.getType())) {
-      U->dump();
-      Val.dump();
+    Type Ty = Val.getType();
+    TypedAttr AllOnesAttr;
+    if (isa<IntegerType>(Ty)) {
+      AllOnesAttr =
+          IntegerAttr::get(Ty, APInt::getAllOnes(Ty.getIntOrFloatBitWidth()));
+    } else if (auto VecTy = dyn_cast<VectorType>(Ty);
+               VecTy && isa<IntegerType>(VecTy.getElementType())) {
+      AllOnesAttr = DenseIntElementsAttr::get(
+          VecTy, APInt::getAllOnes(VecTy.getElementTypeBitWidth()));
+    } else {
+      llvm_unreachable("Bitwise 'not' is supported only on integers or vectors "
+                       "of integers.");
     }
```

Comment:
NIT: if you think it'd improve readability, use `TypeSwitch` here.  LGTM anyway.

Reply:
Good point, tried it out, but in this case (pun intended) it's longer and not significantly better than the if-based version.